### ISSUE TITLE
Inactive Health service overrides /health endpoint of webapp

### ISSUE THREAD

**Issue by vemod** [payara/Payara#2710]
# Inactive Health service overrides /health endpoint of webapp #
----------
Since 4.1.2.181 the disabled Health service still provides /health endpoint which overrides /health endpoint of the webapp which is set to default web module or contextroot=/

* A simple webapp which has /health endpoint and is deployed with --contextroot=/
* Using official payara docker image:
```FROM payara/server-web:4.181```
* Health Service is disabled

Used to work fine in ```payara/server-web:173```. Since ```payara/server-web:4.181``` apps /health endpoint is overwritten by the disabled health service. 

This leads to a lot of problems in microservices environment where /health endpoint is expected to respond with custom response.

## Expected Outcome
No ```/health``` endpoint is available when Health service is disbaled. Or when there is default web module is set in Virtual Domain or deployed with ```--contextroot=/``` the web apps endpoints have precidence.

## Current Outcome
```/health``` endpoint is still active eventough Health service is disbaled

## Steps to reproduce (Only for bug reports) 

1. create a webapp which has /health endpoint.
e.g. mywebapp.war
```
@Controller 
public class HealthController {
        @RequestMapping("/health") @ResponseBody public ResponseEntity<HealthCheckResponse> healthCheckPage() {...}
```
2. make sure /health service is disabled
```
$ asadmin healthcheck-configure --enabled=false
Command healthcheck-configure executed successfully.
$ asadmin get-healthcheck-configuration
Health Check Service Configuration is enabled?: false
```
3. ```asadmin depoy --name mywebapp --contextroot=/ mywebapp.war```
4. Http request to ```/health``` endpoint is handled by the Payara Health Service instead of Mywebapp 
```
$ curl localhost:8080/health
{"outcome":"UP","checks":[]}
```
5. Http request to ```/health/``` endpoint is handled by Mywebapp  
```
$ curl localhost:8080/health/
<?xml version="1.0" encoding="UTF-8" standalone="yes"?><response><meta><message>All OK</message><status>SUCCESS</status><timestamp>2018-05-08T11:35:31.000+02:00</timestamp></meta></response
```
## Environment ##

- **Payara Version**: 4.1.2.181 (Docker payara/server-web:4.181)
- **Edition**: Web
- **JDK Version**: 8 u162 - OpenJDK
- **Operating System**: Linux

## Some details about deployed app ##
![app-screenshot](https://user-images.githubusercontent.com/4331256/39751623-fd37531e-52b8-11e8-8a36-4542db60de81.jpg)

**Comment by vemod** [comment]
btw: I don't have anywhere specified microprofile usages. So the microprofile components are added automatically by Payara. How can I disable them? I thought using ```server-web``` profile should not add microprofile components.

**Comment by mikecroft** [comment]
The issue here is that the `healthcheck-configure` asadmin commands relate to the Payara Health Check Service, **not** MicroProfile HealthCheck.

The [Payara Health Check Service](https://docs.payara.fish/documentation/payara-server/health-check-service/) was introduced a few years ago now and predates the MicroProfile API.

There does seem to be a bug here, though, given that the `/health` endpoint overrides endpoints you specify in your application. I've created `PAYARA-2725` to make the API more configurable and to stop overriding applications.

**Comment by vemod** [comment]
It might be also the case with /metrics endpoint. It is there but I havn't tested if that overrides /metrics from the app. Would be worth to test

**Comment by mikecroft** [comment]
Not a universal one, but each should allow you to enable/disable it (where it makes sense to do so). Some do not make sense to disable since they are inert until you use them, like the Config API. More details are in the docs:
https://docs.payara.fish/documentation/microprofile/

**Comment by smillidge** [comment]
Fixed by #2752
